gate-oauth2/src/main/java/com/netflix/spinnaker/gate/security/oauth2/ExternalAuthTokenFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2025 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.gate.security.oauth2;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.UserInfoRestTemplateFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+import org.springframework.security.oauth2.common.DefaultOAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.provider.authentication.BearerTokenExtractor;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class supports the use case of an externally provided OAuth access token, for example, a
+ * Github-issued personal access token.
+ */
+@Component
+public class ExternalAuthTokenFilter implements Filter {
+
+  @Autowired(required = false)
+  private UserInfoRestTemplateFactory userInfoRestTemplateFactory;
+
+  private BearerTokenExtractor extractor = new BearerTokenExtractor();
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+    Authentication auth = extractor.extract(httpServletRequest);
+    if (auth != null && auth.getPrincipal() != null) {
```

Comment:
The groovy code was:
```
if (auth?.principal) {
```
I'm pretty sure groovy considers the empty string false (see https://blog.mrhaki.com/2009/08/groovy-goodness-tell-groovy-truth.html), so I think we need something like:
```suggestion
    if (auth != null && StringUtils.isNotBlank(auth.getPrincipal()) {
```
where StringUtils is org.apache.commons.lang.StringUtils.

Reply:
https://github.com/spinnaker/gate/pull/1871/commits/aac890fa025f409c229d14069e4ff3ce8ee18963